crates/astria-sequencer/src/fees/query.rs

```diff
@@ -221,204 +241,130 @@ pub(crate) async fn get_fees_for_transaction<S: StateRead>(
-                    .await
-                    .wrap_err("failed to get sudo address change fees")?
-                    .ok_or_eyre(
-                        "fees not found for `SudoAddressChange` action, hence it is disabled",
-                    )?;
+                get_or_init_fees(state, &sudo_address_change_fees).await?;
             }
             Action::IbcSudoChange(_) => {
-                ibc_sudo_change_fees
-                    .get_or_try_init(|| async { state.get_ibc_sudo_change_fees().await })
-                    .await
-                    .wrap_err("failed to get ibc sudo change fees")?
-                    .ok_or_eyre(
-                        "fees not found for `IbcSudoChange` action, hence it is disabled",
-                    )?;
+                get_or_init_fees(state, &ibc_sudo_change_fees).await?;
             }
             Action::Ibc(_) => {
-                ibc_relay_fees
-                    .get_or_try_init(|| async { state.get_ibc_relay_fees().await })
-                    .await
-                    .wrap_err("failed to get ibc relay fees")?
-                    .ok_or_eyre("fees not found for `IbcRelay` action, hence it is disabled")?;
+                get_or_init_fees(state, &ibc_relay_fees).await?;
             }
             Action::IbcRelayerChange(_) => {
-                ibc_relayer_change_fees
-                    .get_or_try_init(|| async { state.get_ibc_relayer_change_fees().await })
-                    .await
-                    .wrap_err("failed to get ibc relayer change fees")?
-                    .ok_or_eyre(
-                        "fees not found for `IbcRelayerChange` action, hence it is disabled",
-                    )?;
+                get_or_init_fees(state, &ibc_relayer_change_fees).await?;
             }
             Action::FeeAssetChange(_) => {
-                fee_asset_change_fees
-                    .get_or_try_init(|| async { state.get_fee_asset_change_fees().await })
-                    .await
-                    .wrap_err("failed to get fee asset change fees")?
-                    .ok_or_eyre(
-                        "fees not found for `FeeAssetChange` action, hence it is disabled",
-                    )?;
+                get_or_init_fees(state, &fee_asset_change_fees).await?;
             }
             Action::FeeChange(_) => {
-                fee_change_fees
-                    .get_or_try_init(|| async { state.get_fee_change_fees().await })
-                    .await
-                    .wrap_err("failed to get fee change fees")?
-                    .ok_or_eyre(
-                        "fees not found for `FeeChange` action, which cannot be disabled",
-                    )?;
+                get_or_init_fees(state, &fee_change_fees).await?;
             }
         }
     }
     Ok(fees_by_asset)
 }
 
-fn calculate_and_add_fees<T: FeeHandler>(
-    act: &T,
+fn calculate_and_add_fees<F: FeeHandler>(
+    action: &F,
     fee_asset: asset::IbcPrefixed,
```

Comment:
since `F` has trait method `fee_asset()` now, I believe we can remove it as an argument